state_synchronizer/src/peer_manager.rs

```diff
@@ -76,6 +79,9 @@ impl PeerManager {
             self.peers
                 .insert(peer_id, PeerInfo::new(true, false, MAX_SCORE));
         }
+
+        self.weighted_index = self.compute_weighted_index();
```

Comment:
nit api change: can we make `compute_weighted_index` to be avoid and just alter internal state. Cause it's called with assignment everywhere anyway

Reply:
also should it be also called for set_peers?

Reply:
I guess it should be? but should only be updated if any of the new peers are alive and upstream